src/Foundation/NSObject2.cs

```diff
@@ -788,6 +788,9 @@ class Observer : NSObject {
 			
 			public Observer (NSObject obj, NSString key, Action<NSObservedChange> cback)
 			{
+				if (cback == null)
+					throw new ArgumentNullException ("cback");
```

Comment:
In this version the parameter name that shows up in the error message doesn't match the name of the parameter in the public method that calls this constructor, so it ends up being confusing for the user.

Maybe you could also change the name of the parameter so that it matches the corresponding name of the caller(s)?

You can also use the `nameof (...)` syntax to avoid hardcoding a string.